packages/noya-renderer/src/primitives.ts

```diff
@@ -163,8 +165,120 @@ export function fill(
 
       break;
     }
-    case Sketch.FillType.Pattern:
-      return paint;
+    case Sketch.FillType.Pattern: {
+      if (!layerFrame || !image) break;
+      const canvasImage = CanvasKit.MakeImageFromEncoded(image);
+      if (!canvasImage) break;
+
+      switch (fill.patternFillType) {
+        case Sketch.PatternFillType.Tile: {
+          paint.setShader(
+            canvasImage.makeShaderCubic(
+              CanvasKit.TileMode.Repeat,
+              CanvasKit.TileMode.Repeat,
+              0,
+              0,
+              CanvasKit.Matrix.multiply(
+                CanvasKit.Matrix.translated(layerFrame.x, layerFrame.y),
+                CanvasKit.Matrix.scaled(
+                  fill.patternTileScale,
+                  fill.patternTileScale,
+                ),
+              ),
+            ),
+          );
+          break;
+        }
+        case Sketch.PatternFillType.Fit: {
+          const bounds = createBounds(layerFrame);
+          const scaledRect = resize(
+            {
+              ...layerFrame,
+              width: canvasImage.width(),
+              height: canvasImage.height(),
+            },
+            layerFrame,
+            'scaleAspectFit',
+          );
+
+          // Scale the largest side to fit, if needed
+          const scale = Math.max(
+            scaledRect.width / canvasImage.width(),
+            scaledRect.height / canvasImage.height(),
+          );
+
+          paint.setShader(
+            canvasImage.makeShaderCubic(
+              CanvasKit.TileMode.Decal,
+              CanvasKit.TileMode.Decal,
+              1,
+              1,
+              CanvasKit.Matrix.multiply(
+                CanvasKit.Matrix.translated(
+                  bounds.midX - scaledRect.width / 2,
+                  bounds.midY - scaledRect.height / 2,
+                ),
+                CanvasKit.Matrix.scaled(scale, scale),
+              ),
+            ),
+          );
+          break;
+        }
+        case Sketch.PatternFillType.Stretch: {
+          paint.setShader(
+            canvasImage.makeShaderCubic(
+              CanvasKit.TileMode.Clamp,
```

Comment:
I noticed the other 2 are `Decal` and this one is `Clamp` - is it different in this case? If not, let's use `Decal` just for consistency